application/common/models/User.php

```diff
@@ -305,7 +305,8 @@ private function validatePassword(): Closure
                     Password::HASH_ALGORITHM,
                     ['cost' => Password::HASH_COST]
                 )) {
-                    $currentPassword->password = $this->password;
+                    $currentPassword->hash = Password::hashPassword($this->password);
+                    $currentPassword->scenario = Password::SCENARIO_REHASH;
```

Comment:
I would use
$currentPassword->setScenario(Password::SCENARIO_REHASH);
Direct access of magical innards scares me. :)

Reply:
I was wondering about that. I saw `setScenario` in the Yii API late in my work here, and did a quick search on this repo. It seemed that a direct assignment to `scenario` is the "norm", but that doesn't make it right.